packages/x-tree-view/src/TreeItem2/TreeItem2.tsx

```diff
@@ -307,7 +375,8 @@ export const TreeItem2 = React.forwardRef(function TreeItem2(
             <TreeItem2Icon status={status} slots={slots} slotProps={slotProps} />
           </IconContainer>
           <Checkbox {...checkboxProps} />
-          <Label {...labelProps} />
+          <LabelInput value={label as string} {...labelInputProps} />
```

Comment:
As discussed in our meeting, I propose to do the following unless it has some drawbacks:
```suggestion
          {status.editing 
            ? <LabelInput {...labelInputProps} />
            : <Label {...labelProps} />
          }  
```

And to remove the `visible` prop from both `LabelInput` and `Label`.